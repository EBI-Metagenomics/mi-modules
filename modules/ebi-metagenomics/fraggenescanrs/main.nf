process FRAGGENESCANRS {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/frag_gene_scan_rs:1.1.0--h4349ce8_0':
        'biocontainers/frag_gene_scan_rs:1.1.0--h4349ce8_0' }"

    input:
    tuple val(meta), path(fasta)
    val(training_file_name)

    output:
    tuple val(meta), path("*.ffn.gz"), emit: nucleotide_fasta
    tuple val(meta), path("*.faa.gz"), emit: amino_acid_fasta
    tuple val(meta), path("*.out.gz"), emit: gene_annotations
    tuple val(meta), path("*.gff.gz"), emit: gff
    path "versions.yml"              , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def catcommand = fasta.name.endsWith(".gz") ? "zcat" : "cat"
    """
    ${catcommand} ${fasta} | FragGeneScanRs \\
        $args \\
        --training-file ${training_file_name} \\
        --thread-num ${task.cpus} \\
        --output-prefix ${prefix}

    gzip *.{ffn,faa,out,gff}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        FragGeneScanRs: \$(FragGeneScanRs --version | sed 's/FragGeneScanRs //')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.faa
    touch ${prefix}.ffn
    touch ${prefix}.out
    touch ${prefix}.gff

    gzip *.{ffn,faa,out,gff}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        FragGeneScanRs: \$(FragGeneScanRs --version | sed 's/FragGeneScanRs //')
    END_VERSIONS
    """
}
