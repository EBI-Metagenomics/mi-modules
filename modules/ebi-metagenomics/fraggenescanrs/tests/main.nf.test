nextflow_process {

    name "Test Process FRAGGENESCANRS"
    script "../main.nf"
    process "FRAGGENESCANRS"

    tag "modules"
    tag "modules_ebimetagenomics"
    tag "fraggenescanrs"

    test("metagenomics-assembly - fasta") {


        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ],
                    file("${moduleDir}/tests/data/example.fa.gz", checkIfExists: true),
                ]
                input[1] = "illumina_5" // train file - this is the default value we use in MGnify for short reads
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("metagenomics-assembly - fasta - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ],
                    file("${moduleDir}/tests/data/example.fa.gz", checkIfExists: true),
                ]
                input[1] = "illumina_5" // train file - this is the default value we use in MGnify for short reads
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
