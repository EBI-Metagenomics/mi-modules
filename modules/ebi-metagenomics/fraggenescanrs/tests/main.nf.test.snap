{
    "metagenomics-assembly - fasta": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.ffn.gz:md5,f5dee7552a6dc7428a153810a8a5fc29"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.faa.gz:md5,041e0629d6cc29699fbb5eda42fdbd9c"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.out.gz:md5,54e3f0864824f17f7a6ea234a9ca7d42"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.gff.gz:md5,cccd3d67491e5d20e9b7df9937a2f386"
                    ]
                ],
                "4": [
                    "versions.yml:md5,9889c439c3439a2cc2d535bab8dbd6e6"
                ],
                "amino_acid_fasta": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.faa.gz:md5,041e0629d6cc29699fbb5eda42fdbd9c"
                    ]
                ],
                "gene_annotations": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.out.gz:md5,54e3f0864824f17f7a6ea234a9ca7d42"
                    ]
                ],
                "gff": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.gff.gz:md5,cccd3d67491e5d20e9b7df9937a2f386"
                    ]
                ],
                "nucleotide_fasta": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.ffn.gz:md5,f5dee7552a6dc7428a153810a8a5fc29"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,9889c439c3439a2cc2d535bab8dbd6e6"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-10T22:43:01.559958"
    },
    "metagenomics-assembly - fasta - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.ffn.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.faa.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.out.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.gff.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "4": [
                    "versions.yml:md5,9889c439c3439a2cc2d535bab8dbd6e6"
                ],
                "amino_acid_fasta": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.faa.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "gene_annotations": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.out.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "gff": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.gff.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "nucleotide_fasta": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.ffn.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,9889c439c3439a2cc2d535bab8dbd6e6"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-10T22:43:49.087012"
    }
}